app/policies/reservation_policy.rb

```diff
@@ -0,0 +1,45 @@
+class ReservationPolicy < ApplicationPolicy
+  def index?
+    true
+  end
+
+  def show?
+    user.id == @record.user_id || user.manager? || user.admin?
+  end
+
+  def create?
+    true
+  end
+
+  def new?
+    create?
+  end
+
+  def update?
+    user.id == @record.user_id || user.manager? || user.admin?
+  end
+
+  def edit?
+    update?
+  end
+
+  def destroy?
+    user.admin?
+  end
+
+  def find_reservations_by_user?
+    user.manager? || user.admin?
+  end
+
+  def find_reservations_by_seance?
+    user.manager? || user.admin?
+  end
+
+  def cancel_reservation?
+    @record.status == "reserved" && (user.id == @record.user_id || user.manager?) || user.admin?
```

Comment:
I understand that cancellation should be only possible if `reservation` has the status `reserved`, but due to my intuition, it's more like a part of the cancellation "process" than a rule to allow/disallow a particular user to perform this action.

Reply:
I removed status check from the policy and implemented it in controller methods